lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -55,7 +62,15 @@ def normalize(os)
     def plain_attributes
       # chef 10 attributes can be access by to_hash directly, chef 11 uses attributes method
       attributes = node.respond_to?(:attributes) ? node.attributes : node.to_hash
-      plainify(attributes.to_hash).flatten.inject(&:merge)
+      if @whitelist
+        attrs = plainify((attributes.select {|key| @whitelist.include?(key)}).to_hash).flatten.inject(&:merge)
+      else
+        attrs = plainify(attributes.to_hash).flatten.inject(&:merge)
+      end
+      if @cachefile
```

Comment:
nitpick, could you please change it to one-liner with statement modifier like this?

```ruby
verify_checksum(attrs) if @cachefile
```